Pod/Classes/Optional Data Sources/VOKCollectionDataSource.m

```diff
@@ -215,27 +247,61 @@ - (void)controllerDidChangeContent:(NSFetchedResultsController *)controller
             
             for (NSDictionary *change in _sectionChanges) {
                 [change enumerateKeysAndObjectsUsingBlock:^(NSNumber *key, id obj, BOOL *stop) {
-                    
+                    NSIndexPath *indexPath, *newIndexPath;
+                    if ([obj isKindOfClass:[NSIndexPath class]]) {
+                        indexPath = obj;
+                    } else if ([obj isKindOfClass:[NSArray class]]) {
+                        indexPath = obj[0];
+                        newIndexPath = obj[1];
+                    }
+
+                    // NSFetchedResultsController doesn't handle fetchLimit
+                    // properly, so we need to check index paths against the
+                    // limit before acting on the change.
+                    NSUInteger fetchLimit = controller.fetchRequest.fetchLimit;
+
                     NSFetchedResultsChangeType type = [key unsignedIntegerValue];
                     switch (type) {
                         case NSFetchedResultsChangeInsert:
-                            [self.collectionView insertSections:[NSIndexSet indexSetWithIndex:[obj unsignedIntegerValue]]];
+                            if (fetchLimit == 0 || indexPath.row < fetchLimit) {
```

Comment:
I guess not: the `NSFetchRequest` documentation doesn't say much about `fetchLimit`. But, since it's an NSUInteger, it defaults to 0, so it's a safe assumption that it's ignored if it were 0 (because otherwise, we wouldn't get getting any results in most cases, since we aren't setting it).